src/main/groovy/org/pih/warehouse/outbound/ImportPackingListItem.groovy

```diff
@@ -0,0 +1,98 @@
+package org.pih.warehouse.outbound
+
+import grails.databinding.BindUsing
+import grails.util.Holders
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.core.Constants
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.core.Person
+import grails.validation.Validateable
+import org.pih.warehouse.inventory.InventoryItem
+import org.pih.warehouse.inventory.ProductAvailabilityService
+import org.pih.warehouse.product.Product
+
+
+class ImportPackingListItem implements Validateable {
+
+    String rowId
+
+    String palletName
+
+    String boxName
+
+    @BindUsing({ obj, source -> Product.findByProductCode(source['product']) })
+    Product product
+
+    String lotNumber
+
+    Date expirationDate
+
+    @BindUsing({ obj, source ->
+        Location location = Location.findByNameOrDefault(source['binLocation'])
```

Comment:
You should look for a bin location name under a specific parent. I think this might be the main reason why you got the performance issues. Bin location name is unique under a specific parent but can be duplicated globally, in this version you can't be sure that you found the bin location that you wanted to find.

Reply:
Well maybe it is not the main reason of performance issues, but definitely looking by name is not enough in terms what you are trying to find here.

Reply:
@awalkowiak yeah, something smelled to me as well. I mean - the performance issues are surely due to the fact that we have not had any index for the name column (because we search for only one location, and searching via id, that has index works fast), but if I understand you correctly, I should look for a location under origin's locations, so:
`origin.getInternalLocation(source['binLocation'])`?

Reply:
Yes this one would be good